modules/kafka/kafka-dest-driver.c

```diff
@@ -144,6 +153,31 @@ _kafka_log_callback(const rd_kafka_t *rkt, int level, const char *fac, const cha
   g_free(buf);
 }
 
+rd_kafka_topic_t *
+kafka_dd_query_insert_topic(KafkaDestDriver *self, const gchar *name)
+{
+  rd_kafka_topic_t *topic;
+  g_mutex_lock(&self->lock);
+  topic = g_hash_table_lookup(self->topic_hash_map, name);
+
+  if(topic)
+    {
+      g_mutex_unlock(&self->lock);
+      return topic;
+    }
+
+  if(kafka_dd_validate_topic_name(name))
+    {
+      topic = rd_kafka_topic_new(self->kafka, g_strdup(name), NULL);
```

Comment:
Maybe this would be better, so there is no code duplication of calling `kafka_dd_validate_topic_name()`, which is already done in `_construct_topic()`:
```
...
topic = _construct_topic(self, name);
if (topic)
  g_hash_table_insert(self->topic_hash_map, g_strdup(name), topic);

g_mutex_unlock(&self->lock);
return topic;
}
```